CMakeLists.txt

```diff
@@ -418,10 +421,97 @@ function(check_working_std_regex result_var_name)
 endfunction()
 check_working_std_regex(SWIFT_HAVE_WORKING_STD_REGEX)
 
-# If SWIFT_HOST_VARIANT_SDK not given, try to detect from the CMAKE_SYSTEM_NAME.
-if(SWIFT_HOST_VARIANT_SDK)
-  set(SWIFT_HOST_VARIANT_SDK_default "${SWIFT_HOST_VARIANT_SDK}")
+#
+# Enable additional warnings.
+#
+swift_common_cxx_warnings()
+
+#
+# Configure SDKs.
+#
+
+function(get_sdk_from_stdlib_target stdlib_target result_var_name)
+  string(REPLACE "-" ";" target_components "${stdlib_target}")
+  list(GET target_components 0 target_sdk)
+  
+  # Darwin SDK names don't match their stdlib-deployment-target names
+  if("${target_sdk}" STREQUAL "macosx")
+    set(target_sdk "OSX")
+  elseif("${target_sdk}" STREQUAL "iphoneos")
+    set(target_sdk "IOS")
+  elseif("${target_sdk}" STREQUAL "iphonesimulator")
+    set(target_sdk "IOS_SIMULATOR")
+  elseif("${target_sdk}" STREQUAL "appletvos")
+    set(target_sdk "TVOS")
+  elseif("${target_sdk}" STREQUAL "appletvsimulator")
+    set(target_sdk "TVOS_SIMULATOR")
+  elseif("${target_sdk}" STREQUAL "watchos")
+    set(target_sdk "WATCHOS")
+  elseif("${target_sdk}" STREQUAL "watchsimulator")
+    set(target_sdk "WATCHOS_SIMULATOR")
+  endif()
+
+  string(TOUPPER "${target_sdk}" target_sdk)
+  set("${result_var_name}" "${target_sdk}" PARENT_SCOPE)
+endfunction()
+
+function(get_arch_from_stdlib_target stdlib_target result_var_name)
+  string(REPLACE "-" ";" target_components "${stdlib_target}")
+  list(GET target_components 1 target_arch)
+  set("${result_var_name}" "${target_arch}" PARENT_SCOPE)
+endfunction()
+
+# Calculate the triple for an sdk/architecture.
+# Will only return a result for supported targets.
+function(get_triple sdk arch result_var_name)
+  if("${sdk}" STREQUAL "LINUX")
+    if("${arch}" STREQUAL "x86_64")
+      set(triple "x86_64-unknown-linux-gnu")
+    elseif("${arch}" STREQUAL "aarch64")
+      set(triple "aarch64-unknown-linux-gnu")
+    elseif("${arch}" STREQUAL "s390x")
+      set(triple "s390x-unknown-linux-gnu")
+    elseif("${arch}" MATCHES "(powerpc64|powerpc64le)")
+      set(triple "${arch}-unknown-linux-gnu")
+    elseif("${arch}" MATCHES "(armv6|armv7)")
+      set(triple "${arch}-unknown-linux-gnueabihf")
+    endif()
```

Comment:
Sure - I think we actually have a vendor flag in the build-script, so we can pass that along and substitute it in.

IIRC, it's important that this exactly matches the LLVM_HOST_TRIPLE from the build-script. It's been a long time since I tested it, but I think that even vendor mismatches cause the tools to not want to co-operate.